### PR TITLE
fixed serialization of oauth2 token scope

### LINKED ISSUE #1929: Refresh token scope list format
### I did this

I send a `POST` to `/oauth2/token` to refresh the access token. The payload looks like this:
```python
{'grant_type': 'refresh_token', 'scope': 'admin email profile displayname preferred_username', 'refresh_token': 'someverylongtoken=='}
```

### I expected the following

A refreshed token to be returned, but I got a 400 and the server log
```
[error]: atr parse error - invalid type: string "admin email profile displayname preferred_username", expected a sequence
```

Am I using the wrong endpoint? I didn't find any other in the documentation. This format seems correct according to spec.

### Kanidm version (and git commit)

kanidmd 1.1.0-alpha.12

### Operating System / Version

official container

### PR DESCRIPTION
**Pull request by Seba-T**
Fixes #1929 

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Seba-T** [comment]
@Firstyear I'm 99% sure we were in the wrong with the syntax we were using, but this is the 'only' thing I found to back it up: https://datatracker.ietf.org/doc/html/rfc6749#section-3.3. The reason I'm not 100% sure is that it feels a bit unlikely that no one stumbled across this issue until now, both the devs and anyone using kani in general.

**Comment by tumbl3w33d** [comment]
In the introspection response you do it [right](https://datatracker.ietf.org/doc/html/rfc7662#section-2.2), btw. That's where I found the space separation first and was confused as well, because I did not expect that. I was talking to kani's API using python's [oauthlib](https://pypi.org/project/oauthlib/) (via flask-dance) and they send the field space separated for refresh_token().

**Comment by Seba-T** [comment]
Yep, I'm seeing now that for the introspection we're doing  `str_join(&scopes)` which indeed joins all the different scopes in a single space separated string. Then I believe no one ever thought about this formatting issue for the refresh token response so this is a legit fix.

File: proto/src/oauth2.rs [hunk 1]
```diff
@@ -77,6 +79,9 @@ pub enum AuthorisationResponse {
     Permitted,
 }
 
+#[serde_as]
+#[skip_serializing_none]
+// this is the equivalent of serde(skip_serializing_if = "Option::is_none") applied to ALL the options
```
Comment (Firstyear) [inline proto/src/oauth2.rs:84]: That's actually cool.

**Review by Firstyear — APPROVED** [review]

File: proto/src/oauth2.rs (hunk 1 above)
Comment (Seba-T) [reply proto/src/oauth2.rs:84]: Yeah I thought about updating all our existing `serde(skip_serialize_if = "Option::is_none)`, but I think that would be better places in its own pr.

**Pull request merged**